api/institutions/authentication.py

```diff
@@ -26,42 +54,50 @@ def authenticate(self, request):
         except (jwt.InvalidTokenError, TypeError):
             raise AuthenticationFailed
 
-        # The JWT `data` payload is expected in the following structure.
-        #
-        # {"provider": {
-        #     "idp": "https://login.circle.edu/idp/shibboleth",
-        #     "id": "CIR",
-        #     "user": {
-        #         "middleNames": "",
-        #         "familyName": "",
-        #         "givenName": "",
-        #         "fullname": "Circle User",
-        #         "suffix": "",
-        #         "username": "user@circle.edu"
-        #     }
-        # }}
         data = json.loads(payload['data'])
         provider = data['provider']
 
         institution = Institution.load(provider['id'])
         if not institution:
             raise AuthenticationFailed('Invalid institution id specified "{}"'.format(provider['id']))
 
-        username = provider['user']['username']
-        fullname = provider['user']['fullname']
+        username = provider['user'].get('username')
+        fullname = provider['user'].get('fullname')
+        given_name = provider['user'].get('givenName')
+        family_name = provider['user'].get('familyName')
+        middle_names = provider['user'].get('middleNames')
+        suffix = provider['user'].get('suffix')
 
-        user, created = get_or_create_user(fullname, username, reset_password=False)
+        # use given name and family name to build full name if not provided
+        if given_name and family_name and not fullname:
+            fullname = given_name + ' ' + family_name
+
+        # institution must provide `fullname`, otherwise we fail the authentication and inform sentry
+        if not fullname:
```

Comment:
Inform Sentry and raise `AuthenticationFailed` exception if `fullname` is not provided or cannot be constructed from `given_name` and `family_name`.